File: C#/0073. Set Matrix Zeroes.cs
public class Solution {
    public void SetZeroes(int[][] matrix) {
        int rowFlag=1;
        int colFlag=1;
        foreach(var num in matrix[0]){
            if(num==0){
                rowFlag=0;
                break;
            }
        }
        
        foreach(var submatrix in matrix){
            if(submatrix[0]==0){
                colFlag=0;
                break;
            }
        }
        
        for(int row=1;row<matrix.Length;row++){
            for(int col=1;col<matrix[0].Length;col++){
                if(matrix[row][col]==0){
                    matrix[0][col]=0;
                    matrix[row][0]=0;
                }
            }
        }
        
        for(int row=1;row<matrix.Length;row++){
            if(matrix[row][0]==0){
                for(int col=0;col<matrix[0].Length;col++)
                    matrix[row][col]=0;
            }
        }
        
        for(int col=1;col<matrix[0].Length;col++){
            if(matrix[0][col]==0){
                for(int row=0;row<matrix.Length;row++)
                    matrix[row][col]=0;
            }
        }
        
        if(matrix[0][0]==0 || rowFlag==0){
            for(int col=1;col<matrix[0].Length;col++)
                matrix[0][col]=0;
        }
        if(matrix[0][0]==0 || colFlag==0){
            for(int row=1;row<matrix.Length;row++)
                matrix[row][0]=0;
        }
        if(rowFlag==0 || colFlag==0)
            matrix[0][0]=0;
        
    }
}
    
    

File: archives/20190519python/0073. Set Matrix Zeroes.py
class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        row_list=set()
        col_list=set()
        for row in range(len(matrix)):
            for col in range(len(matrix[0])):
                if matrix[row][col]==0:
                    row_list.add(row)
                    col_list.add(col)
        
        for row in row_list:
            for col in range(len(matrix[0])):
                matrix[row][col]=0
            
        for col in col_list:
            for row in range(len(matrix)):
                matrix[row][col]=0
